companion/src/shared/timeredit.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) OpenTX
+ *
+ * Based on code named
+ *   th9x - http://code.google.com/p/th9x
+ *   er9x - http://code.google.com/p/er9x
+ *   gruvin9x - http://code.google.com/p/gruvin9x
+ *
+ * License GPLv2: http://www.gnu.org/licenses/gpl-2.0.html
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef TIMEREDIT_H
+#define TIMEREDIT_H
+
+#include <QLineEdit>
+#include <QRegularExpression>
+#include <QRegularExpressionValidator>
+#include <QKeyEvent>
+#include <QWheelEvent>
+#include <QDebug>
+
+class TimerEdit : public QLineEdit
+{
+    Q_OBJECT
+    Q_PROPERTY(int minimumTime READ minimumTime WRITE setMinimumTime)
+    Q_PROPERTY(int maximumTime READ maximumTime WRITE setMaximumTime)
+    Q_PROPERTY(bool showSeconds READ showSeconds WRITE setShowSeconds)
+    Q_PROPERTY(unsigned int singleStep READ singleStep WRITE setSingleStep)
+    Q_PROPERTY(unsigned int pageStep READ pageStep WRITE setPageStep)
+
+
+  public:
+    enum TimeParts {Whole, Polarity, Hours, Minutes, Seconds};
+
+    TimerEdit(QWidget * parent = Q_NULLPTR);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add the "explicit" keyword to this constructor. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1709)